### PR TITLE
Fix PC turning alarm clock off during sleep despite not hearing the alarm

### PR DESCRIPTION
**Pull request by Qrox**
The condition for an alarm clock to wake up the PC in `player_hardcoded_effects.cpp` is different from the condition for a noise to wake up the PC in `sound.cpp`. This is causing the message "You turn off your alarm-clock" to show despite the PC has not waked up. This PR fixes this problem by moving the message code to `sound.cpp`. A message to indicate that the PC missed the alarm is also added so the player will know when it happens.

### CODE REVIEW

File: src/player_hardcoded_effects.cpp [hunk 1]
```diff
@@ -1336,14 +1336,15 @@ void player::hardcoded_effects( effect &it )
                         // 10 minute cyber-snooze
                         it.mod_duration( 100 );
                     }
-                } else {
+                }
+            } else {
+                if( dur <= 1 ) {
+                    add_msg_if_player( m_warning, _( "Your alarm clock rings, but you do not hear it." ) );
```
Comment (kevingranade) [inline src/player_hardcoded_effects.cpp]: Please remove this, we don't notify the player of things they don't percieve.
I could maybe see an argument for it *sometimes* notifying the player that they shut off the alarm but fall back asleep, but that seems like it would be a bit contentious, so it needs to be in another PR.

Comment (Qrox) [inline src/player_hardcoded_effects.cpp]: Sure, will change it later. I was just thinking it could confuse the player if they are not informed of what is really happening.

Comment (Leland) [inline src/player_hardcoded_effects.cpp]: @kevingranade @Qrox I think a simple change to the message would be fine.

E.g. `You check the time - looks like you slept through your alarm.`

**Comment by kevingranade** [comment]
Not so simple, it needs to happen after the player wakes up.

Like I said, if you think this needs to happen, its not out of the
question, but I'd rather not block a bugfix behind a behavioral change, so
move it to another pr.

**Comment by Qrox** [comment]
Probably a new effect can be added to indicate that the player has slept through the alarm, so the message can be displayed when the player wakes up. I'll create a new PR for that once this is merged.

**Comment by tstrimple** [comment]
Not sure about this PR. The effect seems realistic enough to me!

**Comment by kevingranade** [comment]
Had to do a rebase instead of merge, so it didn't auto-close.
